File: data structures and algorithms/cpp/08_004.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <queue>

std::vector<std::string> topKTrendingHashtags(
        std::vector<std::string>& hashtags, int k) {
    // Count frequencies
    std::unordered_map<std::string, int> counter;
    for (const std::string& tag : hashtags) {
        counter[tag]++;
    }
    
    // Use priority queue (min heap)
    auto compare = [&counter](const std::string& a, 
                             const std::string& b) {
        return counter[a] > counter[b];
    };
    
    std::priority_queue<std::string, std::vector<std::string>,
                        decltype(compare)> heap(compare);
    
    for (const auto& pair : counter) {
        heap.push(pair.first);
        if (heap.size() > k) {
            heap.pop();
        }
    }
    
    // Build result vector
    std::vector<std::string> result(k);
    for (int i = k - 1; i >= 0; i--) {
        if (!heap.empty()) {
            result[i] = heap.top();
            heap.pop();
        }
    }
    
    return result;
}

File: data structures and algorithms/cs/08_004.cs
using System;
using System.Collections.Generic;

public class HashtagAnalyzer
{
   public List<string> TopKTrendingHashtags(
       string[] hashtags,
       int k)
   {
      // Count frequencies
      Dictionary<string, int> counter =
          new Dictionary<string, int>();
      foreach (string tag in hashtags)
      {
         if (counter.ContainsKey(tag))
         {
            counter[tag]++;
         }
         else
         {
            counter[tag] = 1;
         }
      }

      // Use priority queue (min heap)
      PriorityQueue<string> heap = new PriorityQueue<string>(
          Comparer<string>.Create(
              (a, b) => counter[a] - counter[b]));

      foreach (string tag in counter.Keys)
      {
         heap.Enqueue(tag);
         if (heap.Count > k)
         {
            heap.Dequeue();
         }
      }

      // Build result list
      List<string> result = new List<string>();
      while (heap.Count > 0)
      {
         result.Insert(0, heap.Dequeue());
      }

      return result;
   }
}